Compile Issue on Generated Models

**Describe the bug**
Build errors with orm generated files on debian 9.

**To Reproduce**
Steps to reproduce the behavior:
1. Create MySQL database structure.
2. Run drogon ctl command to generate 
COMPILE LOG:
```
Scanning dependencies of target hempeno
[  3%] Building CXX object CMakeFiles/hempeno.dir/models/Brands.cc.o
In file included from /usr/include/c++/6/bits/shared_ptr.h:52:0,
                 from /usr/include/c++/6/memory:82,
                 from /usr/local/include/drogon/orm/Result.h:20,
                 from /var/www/html/hempeno/hempeno/models/Brands.h:9,
                 from /var/www/html/hempeno/hempeno/models/Brands.cc:8:
/usr/include/c++/6/bits/shared_ptr_base.h: In instantiation of ‘std::__shared_ptr<_Tp, _Lp>& std::__shared_ptr<_Tp, _Lp>::operator=(std::__shared_ptr<_Tp1, _Lp>&&) [with _Tp1 = long unsigned int; _Tp = long int; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]’:
/usr/include/c++/6/bits/shared_ptr.h:302:4:   required from ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Tp1>&&) [with _Tp1 = long unsigned int; _Tp = long int]’
/var/www/html/hempeno/hempeno/models/Brands.cc:112:40:   required from here
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: error: no matching function for call to ‘std::__shared_ptr<long int, (__gnu_cxx::_Lock_policy)2u>::__shared_ptr(std::remove_reference<std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>&>::type)’
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:1149:7: note: candidate: std::__shared_ptr<_Tp, _Lp>::__shared_ptr(const std::__weak_ptr<_Tp, _Lp>&, std::nothrow_t) [with _Tp = long int; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]
       __shared_ptr(const __weak_ptr<_Tp, _Lp>& __r, std::nothrow_t)
       ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:1149:7: note:   candidate expects 2 arguments, 1 provided
/usr/include/c++/6/bits/shared_ptr_base.h:1097:2: note: candidate: template<class _Alloc, class ... _Args> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_make_shared_tag, const _Alloc&, _Args&& ...)
  __shared_ptr(_Sp_make_shared_tag __tag, const _Alloc& __a,
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:1097:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   candidate expects at least 2 arguments, 1 provided
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:979:17: note: candidate: constexpr std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::nullptr_t) [with _Tp = long int; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u; std::nullptr_t = std::nullptr_t]
       constexpr __shared_ptr(nullptr_t) noexcept : __shared_ptr() { }
                 ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:979:17: note:   no known conversion for argument 1 from ‘std::remove_reference<std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>&>::type {aka std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>}’ to ‘std::nullptr_t’
/usr/include/c++/6/bits/shared_ptr_base.h:976:2: note: candidate: template<class _Tp1> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::auto_ptr<_Up>&&)
  __shared_ptr(std::auto_ptr<_Tp1>&& __r);
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:976:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   ‘std::remove_reference<std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>&>::type {aka std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>}’ is not derived from ‘std::auto_ptr<_Up>’
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:964:2: note: candidate: template<class _Tp1, class _Del, class> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::unique_ptr<_Up, _Ep>&&)
  __shared_ptr(std::unique_ptr<_Tp1, _Del>&& __r)
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:964:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   ‘std::remove_reference<std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>&>::type {aka std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>}’ is not derived from ‘std::unique_ptr<_Tp, _Dp>’
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:951:11: note: candidate: template<class _Tp1> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(const std::__weak_ptr<_Tp1, _Lp>&)
  explicit __shared_ptr(const __weak_ptr<_Tp1, _Lp>& __r)
           ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:951:11: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   ‘std::remove_reference<std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>&>::type {aka std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>}’ is not derived from ‘const std::__weak_ptr<_Tp>’
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:943:2: note: candidate: template<class _Tp1, class> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::__shared_ptr<_Tp1, _Lp>&&)
  __shared_ptr(__shared_ptr<_Tp1, _Lp>&& __r) noexcept
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:943:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:935:7: note: candidate: std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::__shared_ptr<_Tp, _Lp>&&) [with _Tp = long int; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]
       __shared_ptr(__shared_ptr&& __r) noexcept
       ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:935:7: note:   no known conversion for argument 1 from ‘std::remove_reference<std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>&>::type {aka std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>}’ to ‘std::__shared_ptr<long int, (__gnu_cxx::_Lock_policy)2u>&&’
/usr/include/c++/6/bits/shared_ptr_base.h:931:2: note: candidate: template<class _Tp1, class> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(const std::__shared_ptr<_Tp1, _Lp>&)
  __shared_ptr(const __shared_ptr<_Tp1, _Lp>& __r) noexcept
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:931:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:926:7: note: candidate: std::__shared_ptr<_Tp, _Lp>::__shared_ptr(const std::__shared_ptr<_Tp, _Lp>&) [with _Tp = long int; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]
       __shared_ptr(const __shared_ptr&) noexcept = default;
       ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:926:7: note:   no known conversion for argument 1 from ‘std::remove_reference<std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>&>::type {aka std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>}’ to ‘const std::__shared_ptr<long int, (__gnu_cxx::_Lock_policy)2u>&’
/usr/include/c++/6/bits/shared_ptr_base.h:922:2: note: candidate: template<class _Tp1> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(const std::__shared_ptr<_Tp1, _Lp>&, _Tp*)
  __shared_ptr(const __shared_ptr<_Tp1, _Lp>& __r, _Tp* __p) noexcept
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:922:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   candidate expects 2 arguments, 1 provided
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:917:9: note: candidate: template<class _Deleter, class _Alloc> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::nullptr_t, _Deleter, _Alloc)
         __shared_ptr(nullptr_t __p, _Deleter __d, _Alloc __a)
         ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:917:9: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   candidate expects 3 arguments, 1 provided
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:912:2: note: candidate: template<class _Deleter> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::nullptr_t, _Deleter)
  __shared_ptr(nullptr_t __p, _Deleter __d)
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:912:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   candidate expects 2 arguments, 1 provided
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:903:2: note: candidate: template<class _Tp1, class _Deleter, class _Alloc> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(_Tp1*, _Deleter, _Alloc)
  __shared_ptr(_Tp1* __p, _Deleter __d, _Alloc __a)
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:903:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   mismatched types ‘_Tp1*’ and ‘std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>’
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:894:2: note: candidate: template<class _Tp1, class _Deleter> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(_Tp1*, _Deleter)
  __shared_ptr(_Tp1* __p, _Deleter __d)
  ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:894:2: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   mismatched types ‘_Tp1*’ and ‘std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>’
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:884:11: note: candidate: template<class _Tp1> std::__shared_ptr<_Tp, _Lp>::__shared_ptr(_Tp1*)
  explicit __shared_ptr(_Tp1* __p)
           ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:884:11: note:   template argument deduction/substitution failed:
/usr/include/c++/6/bits/shared_ptr_base.h:1011:4: note:   mismatched types ‘_Tp1*’ and ‘std::__shared_ptr<long unsigned int, (__gnu_cxx::_Lock_policy)2u>’
    __shared_ptr(std::move(__r)).swap(*this);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:879:17: note: candidate: constexpr std::__shared_ptr<_Tp, _Lp>::__shared_ptr() [with _Tp = long int; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]
       constexpr __shared_ptr() noexcept
                 ^~~~~~~~~~~~
/usr/include/c++/6/bits/shared_ptr_base.h:879:17: note:   candidate expects 0 arguments, 1 provided
CMakeFiles/hempeno.dir/build.make:141: recipe for target 'CMakeFiles/hempeno.dir/models/Brands.cc.o' failed
make[2]: *** [CMakeFiles/hempeno.dir/models/Brands.cc.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/hempeno.dir/all' failed
make[1]: *** [CMakeFiles/hempeno.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/drogon_ctl/templates/model_cc.csp
+++ b/drogon_ctl/templates/model_cc.csp
@@ -217,13 +217,21 @@ const std::string &[[className]]::getColumnName(size_t index) noexcept(false)
     }
     else if(@@.get<std::string>("rdbms")=="mysql"||@@.get<std::string>("rdbms")=="sqlite3")
     {
+        auto primaryKeyTypeString=@@.get<std::string>("primaryKeyType");
         $$<<"void "<<className<<"::updateId(const uint64_t id)\n";
         $$<<"{\n";
         for(auto col:cols)
         {
             if(col._isAutoVal)
             {
-                $$<<"    _"<<col._colValName<<" = std::make_shared<"<<@@.get<std::string>("primaryKeyType")<<">(static_cast<"<<@@.get<std::string>("primaryKeyType")<<">(id));\n";
+                if(primaryKeyTypeString!="uint64_t")
+                {
+                    $$<<"    _"<<col._colValName<<" = std::make_shared<"<<primaryKeyTypeString<<">(static_cast<"<<primaryKeyTypeString<<">(id));\n";
+                }
+                else
+                {
+                    $$<<"    _"<<col._colValName<<" = std::make_shared<uint64_t>(id);\n";
+                }
                 break;
             }
         }

--- a/drogon_ctl/templates/model_cc.csp
+++ b/drogon_ctl/templates/model_cc.csp
@@ -209,7 +209,7 @@ const std::string &[[className]]::getColumnName(size_t index) noexcept(false)
         }
         $$<<"\n";    
     }
-    if(@@.get<std::string>("rdbms")=="postgresql")
+    if(@@.get<std::string>("rdbms")=="postgresql"||@@.get<int>("hasPrimaryKey")!=1)
     {
         $$<<"void "<<className<<"::updateId(const uint64_t id)\n";
         $$<<"{\n";
@@ -223,7 +223,7 @@ const std::string &[[className]]::getColumnName(size_t index) noexcept(false)
         {
             if(col._isAutoVal)
             {
-                $$<<"    _"<<col._colValName<<" = std::make_shared<uint64_t>(id);\n";
+                $$<<"    _"<<col._colValName<<" = std::make_shared<"<<@@.get<std::string>("primaryKeyType")<<">(static_cast<"<<@@.get<std::string>("primaryKeyType")<<">(id));\n";
                 break;
             }
         }

